pkg/migrations/index.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package migrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/xataio/pgroll/pkg/db"
+)
+
+func createUniqueIndexConcurrently(ctx context.Context, conn db.DB, schemaName string, indexName string, tableName string, columnNames []string) error {
+	quotedQualifiedIndexName := pq.QuoteIdentifier(indexName)
+	if schemaName != "" {
+		quotedQualifiedIndexName = fmt.Sprintf("%s.%s", pq.QuoteIdentifier(schemaName), pq.QuoteIdentifier(indexName))
+	}
+	for retryCount := 0; retryCount < 5; retryCount++ {
+		// Add a unique index to the new column
+		// Indexes are created in the same schema with the table automatically. Instead of the qualified one, just pass the index name.
+		if err := executeCreateUniqueIndexConcurrently(ctx, conn, indexName, schemaName, tableName, columnNames); err != nil {
+			return fmt.Errorf("failed to add unique index %q: %w", indexName, err)
+		}
+
+		// Make sure Postgres is done creating the index
+		isInProgress, err := isIndexInProgress(ctx, conn, quotedQualifiedIndexName)
+		if err != nil {
+			return err
+		}
+
+		ticker := time.NewTicker(500 * time.Millisecond)
+		for isInProgress {
+			<-ticker.C
+			isInProgress, err = isIndexInProgress(ctx, conn, quotedQualifiedIndexName)
+			if err != nil {
+				return err
+			}
+		}
+		ticker.Stop()
```

Comment:
Please move `ticker.Stop` after the line where you initialize it, and add defer:
```golang
ticker := time.NewTicker(500 * time.Millisecond)
defer ticker.Stop()
```

With this change we make sure that if `isIndexInProgress` returns an error, the `ticker` is stopped.